### ISSUE TITLE
Chrome reports ERR_EMPTY_RESPONSE

### ISSUE THREAD

**Issue by douniwan5788** [arendst/Sonoff-Tasmota#4460]
Chrome will report ERR_EMPTY_RESPONSE all the time when using WifiManager on PC or Phone(iOS and Android), Sometimes Safari and Firefox are fine.

_Also, make sure these boxes are checked [x] before submitting your issue - Thank you!_
- [x] _Searched the problem in issues and in the wiki_
- [x] _Hardware used_ : **Sonoff Basic (R2) 2017-10-11**
- [x] _Development/Compiler/Upload tools used_ : **tcpdump**
- [x] _If a pre-compiled release or development binary was used, which one?_ : https://github.com/arendst/Sonoff-Tasmota/releases/download/v6.3.0/sonoff.bin
- [x] _You have tried latest release or development binaries?_ : **Yes**
- [x] _Provide the output of command_``status 0`` :
```
19:03:57 CMD: status 0
19:03:57 RSL: STATUS = {"Status":{"Module":1,"FriendlyName":["Sonoff"],"Topic":"sonoff","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0}}
19:03:57 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"GroupTopic":"sonoffs","OtaUrl":"http://thehackbox.org/tasmota/release/sonoff.bin","RestartReason":"Software/System restart","Uptime":"0T19:51:30","StartupUTC":"2018-11-25T15:12:27","Sleep":0,"BootCount":17,"SaveCount":92,"SaveAddress":"F8000"}}
19:03:57 RSL: STATUS2 = {"StatusFWR":{"Version":"6.3.0","BuildDateTime":"2018-10-30T17:33:53","Boot":31,"Core":"2_3_0","SDK":"1.5.3(aec24ac9)"}}
19:03:57 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["xxx",""],"TelePeriod":300,"SetOption":["0000A003","55818000","00000001"]}}
19:03:57 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":489,"Free":512,"Heap":14,"ProgramFlashSize":1024,"FlashSize":1024,"FlashMode":3,"Features":["00000809","0FDAE794","000183A0","23B617CE","00003BC0"]}}
19:03:57 RSL: STATUS5 = {"StatusNET":{"Hostname":"sonoff-xxx","IPAddress":"192.168.1.4","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer":"119.29.29.29","Mac":"xxx","Webserver":2,"WifiConfig":2}}
19:03:57 RSL: STATUS7 = {"StatusTIM":{"UTC":"Mon Nov 26 11:03:57 2018","Local":"Mon Nov 26 19:03:57 2018","StartDST":"Sun Mar 25 02:00:00 2018","EndDST":"Sun Oct 28 03:00:00 2018","Timezone":8,"Sunrise":"09:25","Sunset":"19:11"}}
19:03:57 RSL: STATUS10 = {"StatusSNS":{"Time":"2018-11-26T19:03:57"}}
19:03:57 RSL: STATUS11 = {"StatusSTS":{"Time":"2018-11-26T19:03:57","Uptime":"0T19:51:30","Vcc":3.174,"POWER":"OFF","Wifi":{"AP":1,"SSId":"xxx","BSSId":"xxx","Channel":11,"RSSI":100}}}
```

**To Reproduce**
Click any blue button in Chrome or just refresh few times

**Expected behavior**
get ERR_EMPTY_RESPONSE

**Screenshots**
![image](https://user-images.githubusercontent.com/3118816/49010148-53a8ae80-f1ad-11e8-9ecf-9012c7cc2fc2.png)
```
19:28:00.151964 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [S], seq 4227056444, win 29200, options [mss 1460,sackOK,TS val 16012117 ecr 0,nop,wscale 4], length 0
19:28:00.155516 IP 192.168.1.4.80 > 192.168.1.1.59862: Flags [S.], seq 15446012, ack 4227056445, win 5840, options [mss 1460], length 0
19:28:00.155724 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [.], ack 1, win 29200, length 0
19:28:00.187901 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [P.], seq 1:416, ack 1, win 29200, length 415: HTTP: GET / HTTP/1.1
19:28:00.208806 IP 192.168.1.4.80 > 192.168.1.1.59862: Flags [P.], seq 1:201, ack 416, win 5425, length 200: HTTP: HTTP/1.1 200 OK
19:28:00.208989 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [.], ack 201, win 30016, length 0
19:28:00.211907 IP 192.168.1.4.80 > 192.168.1.1.59862: Flags [P.], seq 201:1661, ack 416, win 5425, length 1460: HTTP
19:28:00.212070 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [.], ack 1661, win 32120, length 0
19:28:00.218487 IP 192.168.1.4.80 > 192.168.1.1.59862: Flags [P.], seq 1661:2984, ack 416, win 5425, length 1323: HTTP
19:28:00.218645 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [.], ack 2984, win 35040, length 0
19:28:00.590984 IP 192.168.1.1.59862 > 192.168.1.4.80: Flags [F.], seq 416, ack 2984, win 35040, length 0
19:28:00.749788 IP 192.168.1.4.80 > 192.168.1.1.59862: Flags [R.], seq 2984, ack 417, win 5840, length 0
19:28:01.695465 IP 192.168.1.1.59860 > 192.168.1.4.80: Flags [FP.], seq 1973123886:1973124360, ack 15429925, win 29200, length 474: HTTP: GET /cn? HTTP/1.1



19:28:05.415435 IP 192.168.1.1.59856 > 192.168.1.4.80: Flags [FP.], seq 2677924117:2677924565, ack 15397841, win 29200, length 448: HTTP: GET /cn? HTTP/1.1
19:28:06.565431 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [S], seq 2643887742, win 29200, options [mss 1460,sackOK,TS val 16012758 ecr 0,nop,wscale 4], length 0
19:28:06.586014 IP 192.168.1.4.80 > 192.168.1.1.59866: Flags [S.], seq 15462127, ack 2643887743, win 5840, options [mss 1460], length 0
19:28:06.586213 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [.], ack 1, win 29200, length 0
19:28:06.618776 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [P.], seq 1:449, ack 1, win 29200, length 448: HTTP: GET /cn? HTTP/1.1
19:28:06.655489 IP 192.168.1.1.59858 > 192.168.1.4.80: Flags [FP.], seq 366271162:366271636, ack 15413871, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
19:28:07.305441 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [P.], seq 1:449, ack 1, win 29200, length 448: HTTP: GET /cn? HTTP/1.1
19:28:07.598717 IP 192.168.1.4.80 > 192.168.1.1.59866: Flags [F.], seq 1, ack 1, win 5840, length 0
19:28:07.605433 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [.], ack 2, win 29200, length 0
19:28:07.629864 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [F.], seq 449, ack 2, win 29200, length 0
19:28:07.641164 IP 192.168.1.4.80 > 192.168.1.1.59866: Flags [.], ack 1, win 5840, length 0
19:28:07.850574 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [S], seq 3383837508, win 29200, options [mss 1460,sackOK,TS val 16012887 ecr 0,nop,wscale 4], length 0
19:28:07.853327 IP 192.168.1.4.80 > 192.168.1.1.59868: Flags [S.], seq 15478247, ack 3383837509, win 5840, options [mss 1460], length 0
19:28:07.853532 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [.], ack 1, win 29200, length 0
19:28:07.881486 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [P.], seq 1:475, ack 1, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
19:28:08.565417 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [P.], seq 1:475, ack 1, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
19:28:08.685443 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [FP.], seq 1:449, ack 2, win 29200, length 448: HTTP: GET /cn? HTTP/1.1
19:28:08.869198 IP 192.168.1.4.80 > 192.168.1.1.59868: Flags [F.], seq 1, ack 1, win 5840, length 0
19:28:08.875420 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [.], ack 2, win 29200, length 0
19:28:08.907717 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [F.], seq 475, ack 2, win 29200, length 0
19:28:08.915174 IP 192.168.1.4.80 > 192.168.1.1.59868: Flags [.], ack 1, win 5840, length 0
19:28:09.945435 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [FP.], seq 1:475, ack 2, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
19:28:11.455455 IP 192.168.1.1.59866 > 192.168.1.4.80: Flags [FP.], seq 1:449, ack 2, win 29200, length 448: HTTP: GET /cn? HTTP/1.1
19:28:12.715436 IP 192.168.1.1.59868 > 192.168.1.4.80: Flags [FP.], seq 1:475, ack 2, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
19:28:12.775432 IP 192.168.1.1.59860 > 192.168.1.4.80: Flags [FP.], seq 0:474, ack 1, win 29200, length 474: HTTP: GET /cn? HTTP/1.1
```

**Additional context**
I think it's becacase ESP8266WebServer [only accept one simultaneous client](https://github.com/esp8266/Arduino/blob/cd05bae0e8b69eea94adbf153eca4e7c9b96a901/libraries/ESP8266WebServer/src/ESP8266WebServer.cpp#L3), but accept as many TCP connections as possible due to WiFiServer [doesn't has backlog](https://github.com/esp8266/Arduino/issues/5380). modern browsers will always try simultaneous connections  and somehow screwed up esp8266's TCP stack

long cmd like this will also cause ERR_EMPTY_RESPONSE, but I didn't found the reason for now.
```
Timer1 {"Arm":1,"Time":"02:23","Window":0,"Days":"--TW--S","Repeat":1,"Output":1,"Action":1}
```
workaround is break into shorter cmd:
```
Timer1 {"Arm":1,"Time":"02:23","Window":0,}
Timer1 {"Days":"--TW--S","Repeat":1,"Output":1,"Action":1}

```

**Comment by mike2nl** [comment]
@douniwan5788 

Have you cleaned the cache mameory in Chrome?
Which version of Chrome you use?
---------------------------------------------
You have possible not the latest version. 
19:03:57 RSL: STATUS2 = {"StatusFWR":{"Version":"6.3.0","BuildDateTime":"2018-10-30T17:33:53"
---------------------------------------------
You have used 6.3.0 from here?
https://github.com/arendst/Sonoff-Tasmota/releases/tag/v6.3.0
---------------------------------------------
Have you used/tested this version?
http://thehackbox.org/tasmota/release/sonoff-basic.bin

Please, if not make another test and report the status 0 here.

**Comment by andrethomas** [comment]
I cannot reproduce this error using Chrome.

**Comment by ascillato** [comment]
I cannot reproduce this. Working fine for me.

Your issue is not related to the browser. You have another issue. 

Please erase all flash using esptool.py and flash again the precompiled bins from development branch

**Comment by douniwan5788** [comment]
* My chrome version is 70.0.3538.110, and I have even test on different device and platforms (PC/OS X/iOS/Android)
* I use https://github.com/arendst/Sonoff-Tasmota/releases/download/v6.3.0/sonoff.bin which is same as http://thehackbox.org/tasmota/release/sonoff.bin
* just tried http://thehackbox.org/tasmota/release/sonoff-basic.bin, same issue

PS:
```
00:00:00 Project sonoff Sonoff (Topic sonoff, Fallback DVES_321CD7, GroupTopic sonoffs) Version 6.3.0-2_3_0
00:00:00 WIF: WifiManager active for 3 minutes
00:00:00 HTP: Web server active on sonoff-7383 with IP address 192.168.244.1
```
I always get 192.168.244.1 in AP mode, is this right?

flash use these commands with a PL2303 usb2ttl chip on OS X
```
esptool.py --port /dev/tty.usbserial erase_flash
esptool.py --port /dev/tty.usbserial write_flash -fs 1MB -fm dout 0x0 sonoff-basic.bin 
```
and even read back to verify

```
esptool.py --port /dev/tty.usbserial read_flash 0x00000 0x100000 image1M.bin
```

![image](https://user-images.githubusercontent.com/3118816/49022672-5ec10600-f1d0-11e8-8c1a-bd2f17cd4f79.png)
the only different is extra FF

**Comment by ascillato** [comment]
Please try http://thehackbox.org/tasmota/sonoff.bin that is from the development branch (6.3.0.12)

**Comment by ascillato** [comment]
Ok. Thanks for testing. Seems that you are having a network issue. Your router is blocking your sonoff or your firewall is blocking. 

It works fine in my setup. I can connect from several pcs to the same sonoff at the same time without any issue.

Please check your network configuration.

**Comment by ascillato** [comment]
Sorry I can be of more help. I use chrome all the time and never had that issue.

Tasmota is compatible with chrome.

**Comment by Jason2866** [comment]
Yes, working fine using latest Chrome and connected same device from 4 PCs and 1 Tablet and 1 Phone

**Comment by Jason2866** [comment]
Try latest development version. The Sonoff Basic R2 is a problematic device.
I think you have wifi connection issues, as some users with the R2

**Comment by ascillato2** [comment]
Seems that you are having network issues. Please, check that.

Closing this issue as can't be reproduced.

### Support Information
See [Wiki](https://github.com/arendst/Sonoff-Tasmota/wiki) for more information.
See [Chat](https://discord.gg/Ks2Kzd4) for more user experience.

**Comment by douniwan5788** [comment]
I doubt whether it is related to the poor design of Sonoff Basic R2 as @arendst  mentioned in [this post](https://github.com/arendst/Sonoff-Tasmota/issues/3224#issuecomment-405084122)
How can I verify?

**Comment by douniwan5788** [comment]
The problem is solved! I used to set my wifi router Fragmentation Threshold to a very low value. After deleting this setting, everything is working like a charm.

At first, I found that Chrome's tcp packet is larger than Safari because the User-Agent and Accept-Language are longer. After deleting some long headers, Chrome works just like safari, which explains why clicking any blue button calse ERR_EMPTY_RESPONSE (Added Referer increases packet length). After doing some tests with curl, I found that a tcp payload larger than 434 would result in ERR_EMPTY_RESPONSE, so I used the iptables TCPMSS target to limit the MSS to 434, which looks fine. I thought this was a bug in lwip2. But after some digging, I find nothing. I also noticed that there seems to be no MSS issue when sonoff is working in HTTP_MANAGER mode. I decided to change the AP that the sonoff is connected to for tesing. With another wifi router, the problem is disappearing! At that time I realized that the problem was my router. When I tried to add a new SSID for testing, I suddenly found the default fragmentation threshold is 526, which reminds me of the changes I made before.

So the problem is that esp8266 does not detect the true MTU. The only workaround now is to choose between LwIPVariant v2 low memory (TCP_MSS = 536) and v2 high bandwidth (TCP_MSS = 1460) for TCP. But for UDP I have not found a way
